napi-inl.h

```diff
@@ -2242,7 +2242,16 @@ inline PropertyDescriptor::operator const napi_property_descriptor&() const {
 ////////////////////////////////////////////////////////////////////////////////
 
 template <typename T>
-inline ObjectWrap<T>::ObjectWrap() {
+inline ObjectWrap<T>::ObjectWrap(Napi::CallbackInfo callbackInfo) {
+  napi_env env = callbackInfo.Env();
+  napi_value wrapper = callbackInfo.This();
+  napi_status status;
+  napi_ref ref;
+  status = napi_wrap(env, wrapper, this, FinalizeCallback, nullptr, &ref);
```

Comment:
Just want to make sure that it is safe to use `this` instead of passing an instance pointer `T*`.